File: results/dnd-rest-benchmark/lifecycle-runs/20260804T072004Z_claude_claude-sonnet-5_ruby-sinatra/routes/phb.rb
# Player's Handbook rules lookups: spell slots, long rest recovery, and
# carrying capacity. Stateless — no DB access.

post '/v1/phb/spell-slots' do
  body = json_body
  halt 400, { error: 'invalid json' }.to_json if body.nil?

  klass = body['class']
  level = body['level']

  halt 400, { error: 'invalid class' }.to_json unless klass.is_a?(String) && !klass.empty?
  halt 400, { error: 'invalid level' }.to_json unless integerish(level)

  level = level.to_i
  halt 400, { error: "unsupported class/level #{klass}/#{level}" }.to_json unless klass == 'wizard' && WIZARD_SPELL_SLOTS.key?(level)

  { class: klass, level: level, slots: WIZARD_SPELL_SLOTS[level] }.to_json
end

post '/v1/phb/rests/long' do
  body = json_body
  halt 400, { error: 'invalid json' }.to_json if body.nil?

  level = body['level']
  hp_current = body['hp_current']
  hp_max = body['hp_max']
  hit_dice_spent = body['hit_dice_spent']
  exhaustion_level = body['exhaustion_level']

  halt 400, { error: 'invalid level' }.to_json unless integerish(level)
  halt 400, { error: 'invalid hp_current' }.to_json unless integerish(hp_current)
  halt 400, { error: 'invalid hp_max' }.to_json unless integerish(hp_max)
  halt 400, { error: 'invalid hit_dice_spent' }.to_json unless integerish(hit_dice_spent)
  halt 400, { error: 'invalid exhaustion_level' }.to_json unless integerish(exhaustion_level)

  level = level.to_i
  hp_max = hp_max.to_i
  hit_dice_spent = hit_dice_spent.to_i
  exhaustion_level = exhaustion_level.to_i

  max_recovered = [level / 2, 1].max
  new_hit_dice_spent = [hit_dice_spent - max_recovered, 0].max
  new_exhaustion_level = [exhaustion_level - 1, 0].max

  {
    hp_current: hp_max,
    hit_dice_spent: new_hit_dice_spent,
    exhaustion_level: new_exhaustion_level
  }.to_json
end

post '/v1/phb/equipment-load' do
  body = json_body
  halt 400, { error: 'invalid json' }.to_json if body.nil?

  strength = body['strength']
  weight = body['weight']

  halt 400, { error: 'invalid strength' }.to_json unless integerish(strength)
  halt 400, { error: 'invalid weight' }.to_json unless numericish(weight)

  strength = strength.to_i
  capacity = strength * 15

  {
    capacity: capacity,
    weight: weight,
    encumbered: weight > capacity
  }.to_json
end

File: results/dnd-rest-benchmark/lifecycle-runs/20260731T052954Z_claude_claude-sonnet-5_python-flask/dndapp/routes/phb.py
"""Player's Handbook mechanics: spell slots, long rest recovery, encumbrance."""

from flask import Blueprint, jsonify, request

from ..rules import WIZARD_SPELL_SLOTS
from ..validation import valid_int, valid_int_in_range

bp = Blueprint("phb", __name__)


@bp.post("/v1/phb/spell-slots")
def phb_spell_slots():
    data = request.get_json(silent=True) or {}
    char_class = data.get("class")
    level = data.get("level")

    if char_class != "wizard":
        return jsonify(error="unsupported class"), 400
    if not valid_int(level) or level not in WIZARD_SPELL_SLOTS:
        return jsonify(error="unsupported level"), 400

    return jsonify(**{"class": char_class}, level=level, slots=WIZARD_SPELL_SLOTS[level])


@bp.post("/v1/phb/rests/long")
def phb_long_rest():
    data = request.get_json(silent=True) or {}
    level = data.get("level")
    hp_current = data.get("hp_current")
    hp_max = data.get("hp_max")
    hit_dice_spent = data.get("hit_dice_spent")
    exhaustion_level = data.get("exhaustion_level")

    if not valid_int_in_range(level, 1, 20):
        return jsonify(error="invalid level"), 400
    if not valid_int(hp_current) or hp_current < 0:
        return jsonify(error="invalid hp_current"), 400
    if not valid_int(hp_max) or hp_max <= 0:
        return jsonify(error="invalid hp_max"), 400
    if not valid_int(hit_dice_spent) or hit_dice_spent < 0:
        return jsonify(error="invalid hit_dice_spent"), 400
    if not valid_int(exhaustion_level) or exhaustion_level < 0:
        return jsonify(error="invalid exhaustion_level"), 400

    recoverable = max(1, level // 2)
    new_hit_dice_spent = max(0, hit_dice_spent - recoverable)
    new_exhaustion = max(0, exhaustion_level - 1)

    return jsonify(
        hp_current=hp_max,
        hit_dice_spent=new_hit_dice_spent,
        exhaustion_level=new_exhaustion,
    )


@bp.post("/v1/phb/equipment-load")
def phb_equipment_load():
    data = request.get_json(silent=True) or {}
    strength = data.get("strength")
    weight = data.get("weight")

    if not valid_int(strength) or strength < 0:
        return jsonify(error="invalid strength"), 400
    if not isinstance(weight, (int, float)) or isinstance(weight, bool) or weight < 0:
        return jsonify(error="invalid weight"), 400

    capacity = strength * 15
    encumbered = weight > capacity

    return jsonify(capacity=capacity, weight=weight, encumbered=encumbered)
